### PR TITLE
Adds pre-commit prettier hook and runs prettier across the codebase.

### PR DESCRIPTION
**Pull request by mute**
This adds `prettier` along with a precommit hook to run it automatically for `.ts` files being committed.

I added the configuration options to use single quotes and to also add trailing commas for objects and arrays.

It should also integrate really easily with most IDEs; want me to add something to `README.md` somewhere mentioning it as well?

### CODE REVIEW

**Comment by joelgriffith** [comment]
Awesome, this doesn't look that bad, and thanks for adding in the switches.

Not sure if we need to call it out in the README, but maybe the CONTRIBUTING file (I just threw it up recently) would be a better fit. Just a quick heads up type of disclaimer so ppl aren't stunned by the change

File: src/util/ChromeHelper.ts [hunk 1]
```diff
@@ -50,11 +54,15 @@ export class ChromeHelper {
 
     const { tab, targetId } = await chromeUtil.createTab(this.cdp, this.port);
 
-    log(`chrome on ${this.port} launched a new tab at ${targetId}. current tabs: ${this.activeTabs}`);
+    log(
+      `chrome on ${this
+        .port} launched a new tab at ${targetId}. current tabs: ${this
+        .activeTabs}`,
+    );
```
Comment (joelgriffith) [inline src/util/ChromeHelper.ts]: This seems odd. Might be a bug in prettier?

File: src/util/ChromeHelper.ts [hunk 2]
```diff
@@ -66,7 +74,10 @@ export class ChromeHelper {
     this.cdp.Target.closeTarget({ targetId });
     this.activeTabs--;
     this.jobsComplete++;
-    log(`chrome on ${this.port} tab ${targetId} has closed. active tabs: ${this.activeTabs}, completed jobs: ${this.jobsComplete}`);
+    log(
+      `chrome on ${this.port} tab ${targetId} has closed. active tabs: ${this
+        .activeTabs}, completed jobs: ${this.jobsComplete}`,
```
Comment (joelgriffith) [inline src/util/ChromeHelper.ts]: Same thing here

File: src/util/ChromeHelper.ts (hunk 1 above)
Comment (joelgriffith) [inline src/util/ChromeHelper.ts]: nm, the newline won't effect output. Just a jarring first read through

**Comment by joelgriffith** [comment]
ALSO https://github.com/prettier/prettier#badge

**Comment by joelgriffith** [comment]
I pushed up the docs portion of this functionality. I'm not terribly thrilled with the `\n` of the methods, but i'll deal.

**Comment by mute** [comment]
@joelgriffith Yeah, the way it handles long strings isn't ideal -- we could bump up the desired line width setting from the default (80 char) but the prettier docs suggest not doing that. ¯\_(ツ)_/¯

**Pull request merged**